components/twitter/sources/new-follower-of-me/new-follower-of-me.js

```diff
@@ -1,68 +1,57 @@
-const twitter = require('../../twitter.app.js')
+const common = require("../common-followers");
 
 module.exports = {
+  ...common,
   key: "twitter-new-follower-of-me",
   name: "New Follower of Me",
   description: "Emit an event when a user follows you on Twitter",
-  version: "0.0.2",
+  version: "0.0.3",
   props: {
-    db: "$.service.db",
-    twitter,
-    timer: {
-      type: "$.interface.timer",
-      default: {
-        intervalSeconds: 60 * 15,
-      },
+    ...common.props,
+    followersCacheSize: {
+      type: "integer",
+      label: "Followers Cache Size",
+      description: "The maximum amount of follower ID's that will be cached at any given time",
+      min: 10,
+      max: 100,
+      default: 100,
     },
   },
-  async run(event) {
-    const cached = this.db.get("followers") || []
-    const activation = this.db.get("activation") || true
-    let newFollowers = []
-    const followers = (await this.twitter.getFollowers())
-    const latest = [...followers]
-    if (JSON.stringify(latest) === JSON.stringify(cached)) {
-      console.log('No new followers')
-    } else {
-      let lastDeleted
-      let lastGapIndex
-      for (let i = 0; i < cached.length; i++) {
-        if(latest.includes(cached[i])) {
-          delete latest[latest.indexOf(cached[i])]
-          if (i - 1 !== lastDeleted) {
-            //new gap detected
-            lastGapIndex = i
-          }
-          lastDeleted = i
-        }
-      }
+  methods: {
+    ...common.methods,
+    getFollowersCacheSize() {
+      return this.followersCacheSize;
+    },
+    async getRelevantIds() {
+      const isFirstExecution = !this.db.get("hasExecuted");
+      const mostRecentFollowers = this.db.get("followers");
```

Comment:
Small nit.  Can we stick an accessor for this in common_followers?  I don't like that it's a magic string.

Reply:
Makes sense, will do!